monai/apps/pathology/transforms/post/array.py

```diff
@@ -0,0 +1,129 @@
+from monai.utils import TransformBackends, convert_to_numpy, optional_import
+from monai.utils.type_conversion import convert_to_dst_type
+
+label, _ = optional_import("scipy.ndimage.measurements", name="label")
+disk, _ = optional_import("skimage.morphology", name="disk")
+opening, _ = optional_import("skimage.morphology", name="opening")
+watershed, _ = optional_import("skimage.segmentation", name="watershed")
+gaussian, _ = optional_import("skimage.filters", name="gaussian")
+cv2, _ = optional_import("cv2")
+
+__all__ = ["CalcualteInstanceSegmentationMap"]
+
+
+class CalcualteInstanceSegmentationMap(Transform):
+    """
+    Process Nuclei Prediction with XY Coordinate Map.
+
+    Args:
+        threshold_overall: threshold the float values of overall gradient map to int 0 or 1 with specified theashold.
+            Defaults to 0.4.
+        min_size: objects smaller than this size are removed. Defaults to 10.
+        sigma: std. could be a single value, or `spatial_dims` number of values. Defaults to 0.4.
+        kernel_size: the size of the Sobel kernel. Defaults to 21.
+        radius: the radius of the disk-shaped footprint. Defaults to 2.
+
+    """
+
+    backend = [TransformBackends.NUMPY]
+
+    def __init__(
+        self,
+        threshold_overall: float = 0.4,
+        min_size: int = 10,
+        sigma: Union[Sequence[float], float, Sequence[torch.Tensor], torch.Tensor] = 0.4,
+        kernel_size: int = 21,
+        radius: int = 2,
+    ) -> None:
+        self.sigma = sigma
+        self.radius = radius
+        self.kernel_size = kernel_size
+
+        self.remove_small_objects = RemoveSmallObjects(min_size=min_size)
+        self.overall_discreter = AsDiscrete(threshold=threshold_overall)
+        self.fill_holes = FillHoles()
+
+    def __call__(self, seg_pred: NdarrayOrTensor, hover_map: NdarrayOrTensor) -> NdarrayOrTensor:  # type: ignore
+        """
+        Args:
+            seg_pred: the output of the NP(segmentation) branch, shape must be [1, H, W].
+                `seg_pred` have been applied activation layer and must be binarized.
+            hover_map: the horizontal and vertical distances of nuclear pixels to their centres
+                of mass output from the HV branch, shape must be [2, H, W].
+
+        Returns:
+            instance labelled segmentation map with shape [1, H, W].
+
+        Raises:
+            ValueError: when the `seg_pred` dimension is not [1, H, W].
+            ValueError: when the `hover_map` dimension is not [2, H, W].
+
+        """
+        if len(seg_pred.shape) != 3 or len(hover_map.shape) != 3:  # only for 2D
+            raise ValueError("Only support 2D, shape must be [C, H, W]!")
+        if seg_pred.shape[0] != 1:
+            raise ValueError("Only supports single channel segmentation prediction!")
+        if hover_map.shape[0] != 2:
+            raise ValueError("Hover map should be with shape [2, H, W]!")
+
+        pred = convert_to_numpy(seg_pred)
```

Comment:
if `seg_pred` is a metatensor, could you try without this convert?

Reply:
yes, I have removed these two lines and added `convert_to_tensor`, thanks!